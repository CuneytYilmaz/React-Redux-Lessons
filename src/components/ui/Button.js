import React from 'react';
import PropTypes from 'prop-types';

import '../style/Button.css';

const Button = ({ primary, secondary, children, ...otherProps }) => {
  const classNames = ['Button'];

  // Parametre olarak props alınsaydı bu şekilde yazılacaktı:
  //   const { primary, secondary, children, ...otherProps} = props;

  if (primary) {
    classNames.push('Button--primary');
  }

  if (secondary) {
    classNames.push('Button-secondary');
  }

  return (
    <button className={classNames.join(' ')} {...otherProps}>
      {children}
    </button>
  );
};

Button.defaultProps = {
  primary: true,
  secondary: false,
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
};

export default Button;
