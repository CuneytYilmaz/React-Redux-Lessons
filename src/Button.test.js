import React from 'react';
import { shallow } from 'enzyme';

import Button from './components/ui/Button';

it('renders correctly primary button', () => {
  const wrapper = shallow(<Button primary>Button</Button>);

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly secondary button', () => {
  const wrapper = shallow(<Button secondary>Button</Button>);

  expect(wrapper).toMatchSnapshot();
});

it('passes other props to button.', () => {
  const wrapper = shallow(<Button type="submit">Button</Button>);
  expect(wrapper.prop('type')).toBe('submit');
});
