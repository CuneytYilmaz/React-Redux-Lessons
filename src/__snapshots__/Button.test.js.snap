// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly primary button 1`] = `
<button
  className="Button Button--primary"
>
  Button
</button>
`;

exports[`renders correctly secondary button 1`] = `
<button
  className="Button Button--primary Button-secondary"
>
  Button
</button>
`;
